### PR TITLE
Right Click Menu: Add copy/paste/clear style functionality

### LINKED ISSUE #7871: Right Click Menus: Copy/Paste/Clear actions
Being able to copy/paste or clear styles is new to the editor. As such, we're pulling these out of #6137 and #6136 and  #6134 to give them the attention they will need in isolation. 

Per @tmagalhaes-huge[ in slack](https://xwp.slack.com/archives/C01ESBB8ZN1/p1623110387086300): 
Screen Shot 2021-06-08 at 2 17 47 PM

So, to do this - we need to map out what all a "style" is here - are we talking: color, opacity, animation, border? anything else? 

We will also want to add the appropriate unit tests for these new actions and verify that they are keyboard friendly. 

**Outstanding question: Does `Clear image styles` need a keyboard shortcut?** 

These copy/paste/clear actions are present in the shapes menu, the foreground media menu, and the text menu

Screen Shot 2021-06-08 at 2 20 34 PM
https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=7347%3A60835

### PR DESCRIPTION
**Pull request by samwhale**
## Context

[Right Click Context Menu Designs](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=7347%3A60832)

## Summary

Adds functionality for copying, pasting, and clearing styles (resetting to default) from elements. Next sprint we will add this functionality to text and shape elements too.

Note: it already works for all element types. Once the menus are built out we can drop them in and they should work.

## Relevant Technical Choices

Pasting styles can only work between elements of the *same type*.

*Shape*: the `square` shape is the only shape that can have a border and a border radius. Keeping `border` and `borderRadius` as non-clearable attributes so that pasting will not update those styles. 

## To-do

These actions are not in the correct place. Once #8382 is merged, they will be moved to the foreground media element menu

## User-facing changes

<details>
<summary>Gifs showing new functionality</summary>

|Element|Copy/Paste Styles|Clear Styles|
|--|--|--|
|`image`|image|clear-image|
|`video`|video|clear-video|
|`gif`|gif|clear-gif|
|`text`|text|clear-text|
|`shape`|shape|shape-clear|
</details>

## Testing Instructions

Testing this PR will be tedious because you'll have to test each type of element. Those types are
- `image`
- `video`
- `gif`
- `shape`
- `text`

For each element type:
1. Add two different elements to the canvas
2. Update some styles in the inspector panel fo rthat element
3. Right click that element and select `Copy Image Styles` (text to be updated in the future)
4. Right click the other element of the same type. Select `Paste image styles`
5. Should see updated styles on the element and in the inspector
6. Right click either element
7. Select `Clear image styles`
8. Should see styles in table reverted to default values

*Note*: all text types (title 1, title 2, paragraph, etc.) will have the _same_ defaults because they are all different stylings of the same *text* element.

The properties that you will see changed for pasting and clearing are outlined in this table:

|Element type|Properties that are updated when pasting or clearing styles|
|--|--|
|`image`, `video`, and `gif`|-`border`<br/>- `borderRadius`<br/>- `overlay`<br/>- `opacity`<br/>-`flip`<br/>- `rotationAngle`<br/>- `lockAspectRatio`|
|`shape`|- `backgroundColor`<br/>- `opacity`<br/>- `flip`<br/>- `rotationAngle`<br/>- `lockAspectRatio`|
|`text`|- `backgroundColor`<br/>- `backgroundTextMode`<br/>- `border`<br/>- `borderRadius`<br/>- `padding`<br/>- `textAlign`<br/>- `opacity`<br/>-`flip`<br/>- `rotationAngle`<br/>- `lockAspectRatio`|

### QA

- [x] This is a non-user-facing change and requires no QA

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

no

### Does this PR have a legal-related impact?

no

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7871

### CODE REVIEW

File: assets/src/edit-story/app/rightClickMenu/provider.js
Comment (samwhale) [inline assets/src/edit-story/app/rightClickMenu/provider.js]: These don't go here, but I put them there for now until #8382 is merged

File: assets/src/edit-story/app/highlights/quickActions/constants.js
Comment (samwhale) [inline assets/src/edit-story/app/highlights/quickActions/constants.js:29]: moved this to a better place

File: assets/src/edit-story/app/highlights/quickActions/useQuickActions.js
Comment (samwhale) [inline assets/src/edit-story/app/highlights/quickActions/useQuickActions.js:421]: not an element type 🤕

File: assets/src/edit-story/components/canvas/karma/rightClickMenu.karma.js
Comment (samwhale) [inline assets/src/edit-story/components/canvas/karma/rightClickMenu.karma.js:18]: A lot of this is copied form #8382. There are only two new tests, which I'll mark. These are smoke tests since the unit tests cover the actual changed data.

File: assets/src/edit-story/components/canvas/karma/rightClickMenu.karma.js
Comment (samwhale) [inline assets/src/edit-story/components/canvas/karma/rightClickMenu.karma.js]: new tests ✨

File: packages/design-system/src/components/contextMenu/menuItem.js
Comment (samwhale) [inline packages/design-system/src/components/contextMenu/menuItem.js]: This was throwing a proptype error 😕. Create react app uses [svgr](https://react-svgr.com/) under the hood to render svgs, but apparently they're still not a `PropTypes.node`. Feels like `PropTypes.object` is similar to a blanket type like `any` so it'd be good to tighten this up.

File: assets/src/edit-story/app/rightClickMenu/test/useRightClickMenu.js
Comment (samwhale) [inline assets/src/edit-story/app/rightClickMenu/test/useRightClickMenu.js]: temporary until #8382 is merged

**Review by maxyinger — APPROVED** [review]
Works exactly as described. Code looks good too!

I don't want to hold this up for UX reasons, but I did find the current requested UX a little unintuitive once I was able to play around with it here.

I guess I expect this to work like the eye dropper in Adobe Illustrator.

My suggestion as for UX calls here would be:
media: only overlay & opacity
shape: only backgroundColor & opacity
text: only the fields in the `Text` design panel. (font family, line height, color, etc)

As for animations, I think I'd make that a separate copy and past action like `copy element animations` because that should have no problems copy and pasting between fg elements regardless of type also doesn't feel like a `style` to me as a user.

Anyway, my UX 2 cents (if it means anything) is that we should be very selective with what attributes we copy over (only attributes that as a user you'd 100% expect to be copied over) because this action appears very buggy to a user if a property updates that they weren't expecting to update.

I think we should merge this as requested and get it in front of others eyes to get some feedback on how everyone feels and where we wanna take it. Great initial work here!

File: assets/src/edit-story/app/rightClickMenu/constants.js
Comment (swissspidy) [inline assets/src/edit-story/app/rightClickMenu/constants.js]: These words are macOS-specific. What about Windows/Linux?

Why don't we say e.g. "Copy" followed by the keyboard shortcut symbols?

Comment (BrittanyIRL) [reply assets/src/edit-story/app/rightClickMenu/constants.js]: line 68 and line 63 separate mac vs windows/linux and then are set like so: 
```
{
        label: RIGHT_CLICK_MENU_LABELS.COPY_IMAGE_STYLES,
        shortcut: {
          display: isMacOs ? '⌥ ⌘ C' : '⌥ ctrl C',
          title: isMacOs
            ? RIGHT_CLICK_MENU_SHORTCUT_LABELS.OPTION_COMMAND_C
            : RIGHT_CLICK_MENU_SHORTCUT_LABELS.OPTION_CONTROL_C,
        },
      },
````

Is this what you're asking about?

File: assets/src/edit-story/app/rightClickMenu/test/utils.js
Comment (BrittanyIRL) [inline assets/src/edit-story/app/rightClickMenu/test/utils.js:43]: 😂

File: assets/src/edit-story/components/canvas/karma/rightClickMenu.karma.js
Comment (BrittanyIRL) [inline assets/src/edit-story/components/canvas/karma/rightClickMenu.karma.js]: i like this as determining what was copied/pasted, smart!

**Review by BrittanyIRL — APPROVED** [review]
this is awesome! 

I like Max's idea to separate out animation as it's own copy/paste, however this PR gives us a good working foundation to nail down the details since this is probably more complex than design thought it would be. 😅  

Great work, really easy to read through.

File: assets/src/edit-story/app/rightClickMenu/constants.js
Comment (samwhale) [reply assets/src/edit-story/app/rightClickMenu/constants.js:72]: I updated this in #8382, would this be better?

https://github.com/google/web-stories-wp/pull/8382/files#diff-d448253e3cf32f60ad1f345c3bbf87ff388746af5a81e6c1d2b77b758239948fR52-R96

Comment (samwhale) [reply assets/src/edit-story/app/rightClickMenu/constants.js]: > These words are macOS-specific. What about Windows/Linux?
> 
> Why don't we say e.g. "Copy" followed by the keyboard shortcut symbols?

Yeah we do but this is just for the shortcut's aria label. However, we may just remove this in favor of using `kbd` elements for the shortcut labels: 
https://github.com/google/web-stories-wp/pull/8382/files#diff-d448253e3cf32f60ad1f345c3bbf87ff388746af5a81e6c1d2b77b758239948fR52-R96

**Pull request merged**
